src/computability/regular_expressions.lean

```diff
@@ -301,4 +304,46 @@ begin
   exact eq.decidable _ _
 end
 
+omit dec
+
+/-- Map the alphabet of a regular expression. -/
+@[simp] def map (f : α → β) : regular_expression α → regular_expression β
+| 0 := 0
+| 1 := 1
+| (char a) := char (f a)
+| (R + S) := map R + map S
+| (R * S) := map R * map S
+| (star R) := star (map R)
+
+@[simp] lemma map_id : ∀ (P : regular_expression α), P.map id = P
+| 0 := rfl
+| 1 := rfl
+| (char a) := rfl
+| (R + S) := by simp_rw [map, map_id]
+| (R * S) := by simp_rw [map, map_id]
+| (star R) := by simp_rw [map, map_id]
+
+lemma map_comp (g : β → γ) (f : α → β) :
+  ∀ (P : regular_expression α), P.map (g ∘ f) = (P.map f).map g
```

Comment:
The other file states this backwards - it would be good to state either `map_map` or `map_comp` (or both) in both places.